tests/e2e/ctl_v3_auth_test.go

```diff
@@ -984,14 +975,12 @@ func authTestWatch(cx ctlCtx) {
 		var err error
 		if tt.want {
 			err = ctlV3Watch(cx, tt.args, tt.wkv...)
-		} else {
-			err = ctlV3WatchFailPerm(cx, tt.args)
-		}
-
-		if err != nil {
-			if cx.dialTimeout > 0 && !isGRPCTimedout(err) {
+			if err != nil && cx.dialTimeout > 0 && !isGRPCTimedout(err) {
 				cx.t.Errorf("watchTest #%d: ctlV3Watch error (%v)", i, err)
 			}
+		} else {
+			err = ctlV3WatchFailPerm(cx, tt.args)
```

Comment:
@serathius here is one of such cases where no meaningful error output comes due to the watch being cancelled:

>     ctl_v3_auth_test.go:982: [unexpected exit code [5] after running [/home/tjungblu/git/etcd/bin/etcdctl --endpoints=http://localhost:20000 --dial-timeout=7s --user=test-user:pass watch key5 --rev 1]]

Reply:
Makes sense.